server.js

```diff
@@ -684,6 +684,33 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// Coverity Code Advisor On Demand integration
+camp.route(/^\/coverity\/ondemand\/(.+)\/(.+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var badgeType = match[1]; // streams or jobs
+  var badgeTypeId = match[2]; // streamId or jobId
+  var format = match[3];
+  var url = 'https://api.ondemand.coverity.com/' + badgeType + '/' + badgeTypeId + '/badge';
+//  var url = 'https://api-stage01.caas.coverity.com/' + badgeType + '/' + badgeTypeId + '/badge';
+//  var url = 'http://localhost:21020/' + badgeType + '/' + badgeTypeId + '/badge';
+  console.log(url)
+  var badgeData = getBadgeData('coverity', data);
+  request(url, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+      return;
+    }
+    try {
+      badgeData = JSON.parse(buffer);
+      sendBadge(format, badgeData);
```

Comment:
I am responsible for maintaining the json that is sent in response to a "badge" request so I have made it conform to the format used by shields. If the shields json badge format changes I will just update the code on our end to track the change.